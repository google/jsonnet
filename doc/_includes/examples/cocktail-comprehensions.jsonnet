{
  cocktails: {
    "Bee's Knees": {
      // Construct the ingredients by using
      // 4/3 oz of each element in the given
      // list.
      ingredients: [  // Array comprehension.
        { kind: kind, qty: 4 / 3 }
        for kind in [
          'Honey Syrup',
          'Lemon Juice',
          'Farmers Gin',
        ]
      ],
      garnish: 'Lemon Twist',
      served: 'Straight Up',
    },
  } + {  // Object comprehension.
    [sd.name + 'Screwdriver']: {
      ingredients: [
        { kind: 'Vodka', qty: 1.5 },
        { kind: sd.fruit, qty: 3 },
      ],
      served: 'On The Rocks',
    }
    for sd in [
      { name: 'Yellow ', fruit: 'Lemonade' },
      { name: '', fruit: 'Orange Juice' },
    ]
  },
}
