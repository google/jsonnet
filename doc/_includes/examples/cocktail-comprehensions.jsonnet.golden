{
  "cocktails": {
    "Bee's Knees": {
      "garnish": "Lemon Twist",
      "ingredients": [
        {
          "kind": "Honey Syrup",
          "qty": 1.3333333333333333
        },
        {
          "kind": "Lemon Juice",
          "qty": 1.3333333333333333
        },
        {
          "kind": "Farmers Gin",
          "qty": 1.3333333333333333
        }
      ],
      "served": "Straight Up"
    },
    "Screwdriver": {
      "ingredients": [
        {
          "kind": "Vodka",
          "qty": 1.5
        },
        {
          "kind": "Orange Juice",
          "qty": 3
        }
      ],
      "served": "On The Rocks"
    },
    "Yellow Screwdriver": {
      "ingredients": [
        {
          "kind": "Vodka",
          "qty": 1.5
        },
        {
          "kind": "Lemonade",
          "qty": 3
        }
      ],
      "served": "On The Rocks"
    }
  }
}
